integration/spark/shared/src/main/java/io/openlineage/spark/agent/util/RemovePathPatternUtils.java

```diff
@@ -0,0 +1,106 @@
+/*
+/* Copyright 2018-2024 contributors to the OpenLineage project
+/* SPDX-License-Identifier: Apache-2.0
+*/
+
+package io.openlineage.spark.agent.util;
+
+import io.openlineage.client.OpenLineage.InputDataset;
+import io.openlineage.client.OpenLineage.OutputDataset;
+import io.openlineage.spark.api.OpenLineageContext;
+import java.util.List;
+import java.util.Optional;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+import org.apache.commons.lang.StringUtils;
+
+/**
+ * Utility class to handle removing path patterns in dataset names. Given a configured regex pattern
+ * with "remove" group defined, class methods run regex replacements on all the datasets available
+ * within the event
+ */
+public class RemovePathPatternUtils {
+  public static final String REMOVE_PATTERN_GROUP = "remove";
+  public static final String SPARK_OPENLINEAGE_DATASET_REMOVE_PATH_PATTERN =
+      "spark.openlineage.dataset.removePath.pattern";
+
+  public static List<OutputDataset> removeOutputsPathPattern(
```

Comment:
Maybe `List<? extends Dataset>` and implement this generically with passed dataset builder? Nit, do what you think is best.

Reply:
I have tried this approach but did not succeed. 
Creating `OutputDataset` and `InputDataset` differs significantly and these have to be two separate methods. 
The iterating part, that goes through the list, is common.
So a common generic method would need to call methods  per `OutputDataset` or `InputDataset`, which didn't look as an improvement to me.